src/app/events/slash-create/ComponentInteraction.ts

```diff
@@ -0,0 +1,48 @@
+import { DiscordEvent } from '../../types/discord/DiscordEvent';
+import { ComponentContext } from 'slash-create';
+import Log, { LogUtils } from '../../utils/Log';
+import componentInteractionSquadUp from './squad/ComponentInteractionSquadUp';
+
+export interface ComponentMeta {
+	feature: string,
+	method:string,
+	label:string,
+	data?:string
+}
+
+export default class implements DiscordEvent {
+
+	name = 'componentInteraction';
+	once = false;
+
+	formatComponentMetaData(componentContext: ComponentContext): ComponentMeta {
+
+		const componentMeta = {} as ComponentMeta;
+		try {
+			[componentMeta.feature, componentMeta.method, componentMeta.label, componentMeta.data] = componentContext.customID.split(':');
+		} catch (e) {
+			// do nothing 
+		}
+
+		return componentMeta;
+
+	}
+
+	async execute(componentContext: ComponentContext): Promise<any> {
+
+		if (componentContext.componentType === 2) {
+			await componentContext.defer(true);
+			Log.debug('component interaction metadata');
```

Comment:
Would this be better as `debug` or `info`?

Reply:
Both of these logs are really supposed to be one message only. I just didn't know if the logger can do that. This log would be the message and next Log would be the data. Together I would say they should be debug.

Reply:
This seems to be a situation where it would be helpful to run with a logger like `Pino` or `Winston`; something much more fully featured.